database/src/main/postgres/runs/V1.9.7__create_partitioning.sql

```diff
@@ -0,0 +1,107 @@
+    IN  i_partitioning          JSONB,
+    IN  i_by_user               TEXT,
+    IN  i_parent_partitioning   JSONB = NULL,
+    OUT status                  INTEGER,
+    OUT status_text             TEXT,
+    OUT id_partitioning         BIGINT
+) RETURNS record AS
+$$
+    -------------------------------------------------------------------------------
+--
+-- Function: runs.create_partitioning(3)
+--      Creates a partitioning entry
+--
+-- Parameters:
+--      i_partitioning          - partitioning to create or which existence to check
+--      i_by_user               - user behind the change
+--      i_parent_partitioning   - parent partitioning of the provided partitioning, optional
+--
+-- Returns:
+--      status              - Status code
+--      status_text         - Status text
+--      id_partitioning     - id of the partitioning
+--
+-- Status codes:
+--      11                  - Partitioning created
+--      12                  - Partitioning parent registered
+--      31                  - Partitioning already present
+--
+-------------------------------------------------------------------------------
+DECLARE
+    _fk_parent_partitioning BIGINT := NULL;
+    _create_partitioning    BOOLEAN;
+    _status                 BIGINT;
+BEGIN
+
+    id_partitioning := runs._get_id_partitioning(i_partitioning, true);
+
+    _create_partitioning := id_partitioning IS NULL;
+
+    IF i_parent_partitioning IS NOT NULL THEN
+        SELECT CPINE.id_partitioning
+        FROM runs.create_partitioning_if_not_exists(i_parent_partitioning, i_by_user, NULL) AS CPINE
+        INTO _fk_parent_partitioning;
+    END IF;
+
+
+    IF _create_partitioning THEN
+        INSERT INTO runs.partitionings (partitioning, created_by)
+        VALUES (i_partitioning, i_by_user)
+        RETURNING partitionings.id_partitioning
+            INTO create_partitioning.id_partitioning;
+
+        PERFORM 1
+        FROM flows._create_flow(id_partitioning, i_by_user);
+
+        status := 11;
+        status_text := 'Partitioning created';
+    ELSE
+        status := 31;
+        status_text := 'Partitioning already present';
+        RETURN;
+    END IF;
+
+    IF i_parent_partitioning IS NOT NULL THEN
+
+        SELECT ATPF.status
+        FROM flows._add_to_parent_flows(_fk_parent_partitioning, id_partitioning, i_by_user) AS ATPF
+        INTO _status;
+
+        IF _create_partitioning THEN
+            -- copying measure definitions to establish continuity
+            INSERT INTO runs.measure_definitions(fk_partitioning, measure_name, measured_columns, created_by, created_at)
+            SELECT id_partitioning, CMD.measure_name, CMD.measured_columns, CMD.created_by, CMD.created_at
+            FROM runs.measure_definitions CMD
+            WHERE CMD.fk_partitioning = _fk_parent_partitioning;
+
+            -- additional data are not copied, they are specific for particular partitioning
+        ELSIF (_status = 11) THEN
```

Comment:
**Semi-important:**
This will change _status -> 11, Partitioning created_ too to status -> 12
Do we want that? 
Or maybe three OK statuses, 11, 12 and 13 fore paritioning created and linked to parent? 🤔